numpy/core/src/umath/simd.inc.src

```diff
@@ -27,6 +27,19 @@
 #include <stdlib.h>
 #include <string.h> /* for memcpy */
 
+/* Figure out the right abs function for pointer addresses */
+static NPY_INLINE npy_intp
+abs_intp(npy_intp x)
+{
+#if (NPY_SIZEOF_INTP <= NPY_SIZEOF_INT)
+    return abs(x);
+#elif (NPY_SIZEOF_INTP <= NPY_SIZEOF_LONG)
+    return labs(x);
+#else
+    return llabs(x);
+#endif
+}
+
```

Comment:
Hmm, certainly no worse than it was ;) I'm thinking about the corner case of `abs` not working for the most negative value of signed integers, but that is a whole other can of worms.